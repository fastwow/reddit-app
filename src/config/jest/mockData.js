export const postsResponse = {
  children: [
    {
      data: {
        url: 'https://v.redd.it/1zhxx58rhp421',
        id: '98vj9e',
        thumbnail: 'https://b.thumbs.redditmedia.com/NFJ5zzCUnxTgAl71EkjSSCePc7wtcCLqTdbYgrwJkdU.jpg',
        title: 'Blinks of affection',
      },
    },
    {
      data: {
        url: 'https://i.imgur.com/GsoJi0d.gifv',
        id: 'a6q94b',
        thumbnail: 'https://b.thumbs.redditmedia.com/cjYmtjKiy9OT1y43RFMuTkcH_WmTM66RsSuN7c6xkhA.jpg',
        title: 'This church looks like a gingerbread house.',
      },
    },
  ],
};

export const posts = postsResponse.children.map(c => c.data);
