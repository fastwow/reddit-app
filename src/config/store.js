/* global __DEV__ */
import {createStore, applyMiddleware, compose} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer from '../reducers';
import storage from 'redux-persist/lib/storage';
import {persistReducer} from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

let middleware = [thunk];

if (__DEV__) {
  const reduxImmutableStateInvariant = require('redux-immutable-state-invariant').default();
  middleware = [...middleware, reduxImmutableStateInvariant, logger];
} else {
  middleware = [...middleware];
}

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['favorites'],
  debug: __DEV__,
  stateReconciler: autoMergeLevel2,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default function configureStore(initialState) {
  const enhancer = compose(
    applyMiddleware(...middleware),
  );
  return createStore(persistedReducer, initialState, enhancer);
}
