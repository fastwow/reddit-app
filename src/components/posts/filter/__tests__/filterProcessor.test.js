import applyFilter from '../filterProcessor';
import {posts} from '../../../../config/mockData';

describe('filterProccessor', () => {
  it('Filter applied correctly', () => {
    expect(applyFilter(posts, {searchTerm: 'house'})).toMatchSnapshot();
  });
});
