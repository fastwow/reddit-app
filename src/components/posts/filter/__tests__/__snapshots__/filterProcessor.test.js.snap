// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterProccessor Filter applied correctly 1`] = `
Array [
  Object {
    "id": "a6q94b",
    "thumbnail": "https://b.thumbs.redditmedia.com/cjYmtjKiy9OT1y43RFMuTkcH_WmTM66RsSuN7c6xkhA.jpg",
    "title": "This church looks like a gingerbread house.",
    "url": "https://i.imgur.com/GsoJi0d.gifv",
  },
]
`;
